backend/src/nodes/nodes/image_utility/convolve.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import numpy as np
+
+from . import category as ImageUtilityCategory
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import ImageInput, NumberInput, TextInput
+from ...properties.outputs import ImageOutput
+from ...properties import expression
+
+
+@NodeFactory.register("chainner:image:image_convolve")
+class ImageConvolveNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Convolves input image with input kernel (kernel values separated by commas)"
+        self.inputs = [
+            ImageInput(),
+            TextInput("Kernel String"),
+            NumberInput("Kernel Width/Height", minimum=0, default=3),
```

Comment:
Regarding the kernel format: Maybe we could use a format that encodes the kernel dimensions as well as its weights? With a text area inputs, we could use a format like this:
```
1   0 -1
2.5 0 -2.5
1   0 -1
```
Newlines are used as row separators and spaces are used as column separators.

This format should be rather intuitive for new users, and it fully describes the kernel. The latter is very useful, because it will us to make a specialized kernel input in the future (e.g. as a popover as suggested on discord).

Reply:
I'm fine with this so long as we have a node that can create kernels without having to format things like this (or i guess using the popover would be fine like you said)

Generally though i think I'd prefer using separate nodes for these kinds of things though, that way other numbers can be connected to the node in order to create them

Reply:
Yes, this string representation was mostly meant as an internal string format for kernels that is also usable right now. Eventually, we should have a dedicated kernel input that creates these strings.

>  other numbers can be connected to the node in order to create them

Do you really want to have an e.g. 3x3 kernel node with 9 number inputs? I would have said that we treat kernels as literals and then support math operations on them, so people can create more complex kernels from simpler ones. Well, I guess we don't have to iron out the specifics right now.

Reply:
Regarding the current string representation, there doesn't seem to be a way to have multiline string input. I have made a version of the kernel input to support this multiline input though which looks like this:
```python
kernel = np.stack([l.split() for l in kernel_in.splitlines()], axis=0).astype(float)
```
and works for inputs that look like this:
```yaml
integer only input:
1 0 -1
2 0 -2
1 0 -1

irregular input:
1   0  -1
2.5  0   -2.5
1    0  -1
```
But does not work in chaiNNer due to the lack of support for multiline text input. I'm not sure what to do from here.

Reply:
> there doesn't seem to be a way to have multiline string input

We have [NoteTextAreaInput](https://github.com/chaiNNer-org/chaiNNer/blob/3fae92c05ffc94be5e9b84c0abcffa084bd6cff6/backend/src/nodes/properties/inputs/generic_inputs.py#L225). This input is only used in the Note nodes right now, so we might encounter a few bugs since it hasn't really been tested a lot. But this is actually a good thing, because we can then fix those bugs in later PRs. For now, let's just use that input, bugy or not, and fix anything that comes up later.